javagen/src/main/java/com/azure/autorest/model/clientmodel/ClientMethod.java

```diff
@@ -93,6 +93,8 @@ public class ClientMethod {
 
     private JavaVisibility methodVisibility;
 
+    private boolean implementationOnly = false;
```

Comment:
`implementationOnly` might not be a good name, or it might need to be an enum.

And it kind of group with `methodVisibility` above, but still not same or mergeable. (as we can have public method in implementation, or package private method in implementation).